### DISCUSSION TITLE
TimberPost equivalent in Timber 2.x

### DISCUSSION THREAD

**Discussion by robpetrin** [timber/timber#2814] [Q&A] [answered]
Hi! I'm responsible for managing a number of websites using Timber/Twig which I wasn't involved in developing. In preparation for the PHP 8.2 upgrade, I'm attempting to shift these websites from using the plugin version 1.x to retrofitting 2.x within their themes.

Most instances of Timber in a template start with this syntax

`$context = Timber::get_context();
$context[ 'footer' ] = new TimberPost();`

It seems that TimberPost() is not a thing in 2.x -- does anyone know what the equivalent is? I see references to this [in the 2.x docs](https://timber.github.io/docs/v2), but nothing there really resonates with the code I'm looking at, or are explicit how to update this kind of assignment for 2.x

**Comment by verma-s3** [comment, marked as answer]
use this: 
```$context = Timber::context(); $context['footer'] = Timber::get_post();```
You can find your guidelines for this code in the [2.x docs here](https://timber.github.io/docs/v2/upgrade-guides/2.0/#an-updated-api-to-get-timber-objects).

**Reply by nlemoine** [reply]
Since `post` is set in context whenever available, you also can do:
```php
$context = Timber::context();
$context['footer'] = $context['post'] ?? null;
```
